### PR TITLE
Add new protobufs for schedules-in-environments.

### PR DESCRIPTION
**Pull request by BenBirt**
for https://github.com/dataform-co/dataform-co/issues/5333

### CODE REVIEW

File: protos/core.proto [hunk 1]
```diff
@@ -26,22 +26,62 @@ message ProjectConfig {
...
-message Environments {
-  repeated Environment environments = 1;
+    message EmailNotificationChannel {
+      repeated string to = 1;
+    }
+    message SlackNotificationChannel {
+      string webhook_url = 1;
+    }
+    oneof channel {
+      EmailNotificationChannel email = 2;
```
Comment (GJMcGowan) [inline protos/core.proto:80]: Out of interest, why 2/3 here and not 1/2? Seems scoped

Comment (BenBirt) [reply protos/core.proto:80]: Fields inside a `oneof` are actually embedded in the `oneof`'s parent Message (a `oneof` is really just enforced at serialization/deserialization time). Here the parent Message's tag `1` is already used.

File: protos/core.proto [hunk 2]
```diff
@@ -26,22 +26,62 @@ message ProjectConfig {
   string gcloud_project_id = 6;
 }
 
-message Environment {
-  string name = 1;
+message Environments {
+  message Environment {
```
Comment (lewish) [inline protos/core.proto:30]: I'm not a fan of this, it's going to make all our imports/Types horribly long for no obvious benefit?
Would it not be better to put this in a new namespace?

File: protos/core.proto [hunk 3]
```diff
@@ -26,22 +26,62 @@ message ProjectConfig {
...
+      string name = 1;
+      string cron = 2;
+      repeated string tags = 3;
   
-  message GitReference {
-    oneof git_reference {
-      string branch = 1;
-      string commit_sha = 2;
+      // Optional.
+      bool include_dependencies = 4;
```
Comment (lewish) [inline protos/core.proto:50]: Why not put this stuff in a new type?
It's basically a stripped down RunConfig, I understand why we don't use that directly here but will be easier to keep in sync if we have seperate messages

File: protos/core.proto [hunk 4]
```diff
@@ -26,22 +26,62 @@ message ProjectConfig {
...
+      bool full_refresh = 5;
+      bool disabled = 6;
+  
+      message NotificationRequirement {
```
Comment (lewish) [inline protos/core.proto:54]: ditto, writing `dataform.Environments.Environment.Schedule.NotificationRequirement` is going to be annoyingly verbose.
can we make Schedule a top level thing?
I think a new namespace would help a lot here

File: protos/core.proto (hunk 2 above)
Comment (BenBirt) [reply protos/core.proto:30]: We can put it in a different namespace, that's fine with me, but I do think it was a mistake not to nest `Environment` inside `Environments` in the first place. 

For me at least, nesting messages inside the scope where they're used feels much cleaner, and that makes it more readable/understandable.

I actually wish that more of these protobufs were nested, I often find it hard to navigate our protobuf files and figure out what messages are used where.

Comment (BenBirt) [reply protos/core.proto:30]: Also I don't really think we should structure protobufs to suit any specific target language. If our TS types become verbose, there are workarounds for that, e.g. type aliases.

File: protos/core.proto (hunk 3 above)
Comment (BenBirt) [reply protos/core.proto:50]: https://github.com/dataform-co/dataform/pull/722

**Pull request merged**